test/realm_set_test.dart

```diff
@@ -0,0 +1,535 @@
+  late String make;
+}
+
+@RealmModel()
+class _TestRealmSets {
+  @PrimaryKey()
+  late int key;
+
+  late Set<bool> boolSet;
+  late Set<int> intSet;
+  late Set<String> stringSet;
+  late Set<double> doubleSet;
+  late Set<DateTime> dateTimeSet;
+  late Set<ObjectId> objectIdSet;
+  late Set<Uuid> uuidSet;
+  late Set<RealmValue> mixedSet;
+  late Set<_Car> objectsSet;
+
+  late Set<bool?> nullableBoolSet;
+  late Set<int?> nullableIntSet;
+  late Set<String?> nullableStringSet;
+  late Set<double?> nullableDoubleSet;
+  late Set<DateTime?> nullableDateTimeSet;
+  late Set<ObjectId?> nullableObjectIdSet;
+  late Set<Uuid?> nullableUuidSet;
+
+  /// When changing update also `supportedTypes`
+  Sets setByType(Type type) {
+    switch (type) {
+      case bool:
+        return Sets(boolSet as RealmSet<bool>, [true, false]);
+      case int:
+        return Sets(intSet as RealmSet<int>, [-1, 0, 1]);
+      case String:
+        return Sets(stringSet as RealmSet<String>, ['Tesla', 'VW', 'Audi']);
+      case double:
+        return Sets(doubleSet as RealmSet<double>, [-1.1, 0.1, 1.1, 2.2, 3.3, 3.14]);
+      case DateTime:
+        return Sets(dateTimeSet as RealmSet<DateTime>, [DateTime(2023).toUtc(), DateTime(1981).toUtc()]);
+      case ObjectId:
+        return Sets(objectIdSet as RealmSet<ObjectId>, [ObjectId.fromTimestamp(DateTime(2023).toUtc()), ObjectId.fromTimestamp(DateTime(1981).toUtc())]);
+      case Uuid:
+        return Sets(uuidSet as RealmSet<Uuid>, [Uuid.fromString("12345678123456781234567812345678"), Uuid.fromString("82345678123456781234567812345678")]);
+      case RealmValue:
+        return Sets(mixedSet as RealmSet<RealmValue>, [RealmValue.nullValue(), RealmValue.bool(true), RealmValue.int(1), RealmValue.string("text")]);
+      case RealmObject:
+        return Sets(objectsSet as RealmSet<Car>, [Car("Tesla"), Car("VW"), Car("Audi")], (realm, value) => realm.find<Car>((value as Car).make));
+      case _NullableBool:
+        return Sets(nullableBoolSet as RealmSet<bool?>, [...setByType(bool).values, null]);
+      case _NullableInt:
+        return Sets(nullableIntSet as RealmSet<int?>, [...setByType(int).values, null]);
+      case _NullableString:
+        return Sets(nullableStringSet as RealmSet<String?>, [...setByType(String).values, null]);
+      case _NullableDouble:
+        return Sets(nullableDoubleSet as RealmSet<double?>, [...setByType(double).values, null]);
+      case _NullableDateTime:
+        return Sets(nullableDateTimeSet as RealmSet<DateTime?>, [...setByType(DateTime).values, null]);
+      case _NullableObjectId:
+        return Sets(nullableObjectIdSet as RealmSet<ObjectId?>, [...setByType(ObjectId).values, null]);
+      case _NullableUuid:
+        return Sets(nullableUuidSet as RealmSet<Uuid?>, [...setByType(Uuid).values, null]);
+      default:
+        throw RealmError("Unsupported type $type");
+    }
+  }
+
+  List<Object?> values(Type type) {
+    return setByType(type).values;
```

Comment:
I believe we have discussed this, but again.  Do you see an issue with this? Is it a performance concern we should be addressing

Reply:
I also found it confusing when I read it - I think a more straightforward approach would be to create `setByType(type)` in your test and then call `.set` and `.values` on it.

Reply:
These tests are combing multiple concepts into one. `TestRealmSets` can be both unmanaged and managed and it contains RealmSet of different types so in order this to work in all cases one needs to go through the concrete instance of `TestRealmSets` every time to get the correct set and value instances.
so each test needs to call 
```
var set = testSet.setByType(type).set;
var values = testSet.values(type);
```
each time it needs to work with the correct data. 

Also `testSet.values` is just a convenience method for `testSet.setByType(type).values`.

I am not getting an local var of the `testSet.setByType(type)` result since then I will need three code lines each time to get the `set` and `values` from that local var. And I just don't care too much of the extra object created since this will get collected by the GC as soon as the method exits, since its a gen0 object. 

In general I agree these auto generated tests in all test files in the project that we are creating are hard to work with. Especially amplified by the lack of good tooling support for such tests. But then I don't think we have an option not to auto generate these. 

I will go with the tests that we have here and handle the feedback for the missing tests coverage.